--- isdnutils-3.9.20060704/capifax/capifax.c.orig	2008-03-19 08:39:42.046270120 +0100
+++ isdnutils-3.9.20060704/capifax/capifax.c	2008-03-19 08:37:00.733836759 +0100
@@ -244,6 +244,8 @@
 	FileTransfer = FALSE;
 	fclose(f);
 	switch (reason) {
+		case 0x3400: /* Normal termination, no reason available */
+		case 0x3480: /* Normal termination */
 		case 0x3490: /* Normal call clearing */
 		case 0x349f: /* Normal, unspecified */
 			return reason_b3;
--- isdnutils-3.9.20060704/capifax/c20msg.c.orig	2008-03-19 08:39:17.085893598 +0100
+++ isdnutils-3.9.20060704/capifax/c20msg.c	2008-03-19 06:40:41.442747376 +0100
@@ -151,6 +151,12 @@
 		case 0x3319:
 			return "Illegal parameter coding (e.g. SFF coding error)";
 			/* disconnect causes from the network according to ETS 300 102-1/Q.931 */
+		case 0x3400:
+			return "Normal termination, no reason available";
+            /* see: http://www.capisuite.de/de-manual/apb.html */
+		case 0x3480:
+			return "Normal termination";
+            /* see: http://www.capisuite.de/de-manual/apb.html */
 		case 0x3481:
 			return "Unallocated (unassigned) number";
 		case 0x3482:
