Index: isdnutils-3.9.20060704+dfsg.1/isdnlog/isdnrep/isdnbill.c
===================================================================
--- isdnutils-3.9.20060704+dfsg.1.orig/isdnlog/isdnrep/isdnbill.c	2010-07-27 21:23:04.569878770 +0200
+++ isdnutils-3.9.20060704+dfsg.1/isdnlog/isdnrep/isdnbill.c	2010-07-27 21:23:48.913880614 +0200
@@ -856,7 +856,7 @@
 #ifdef AK
   auto     FILE    *f = fopen("/www/log/isdn.log", "r");
 #else
-  auto     FILE    *f = fopen("/var/log/isdn.log", "r");
+  auto     FILE    *f = fopen("/var/lib/isdn/calls", "r");
 #endif
   auto     char     s[BUFSIZ], sx[BUFSIZ];
   auto     int      i, l, col, day, lday = UNKNOWN, month, lmonth = UNKNOWN;
