diff -urNad isdnutils-3.9.20060704~/eurofile/scripts/eftd.sh.in isdnutils-3.9.20060704/eurofile/scripts/eftd.sh.in
--- isdnutils-3.9.20060704~/eurofile/scripts/eftd.sh.in	2008-04-04 11:41:12.000000000 +0200
+++ isdnutils-3.9.20060704/eurofile/scripts/eftd.sh.in	2008-04-04 11:41:58.000000000 +0200
@@ -1,5 +1,12 @@
 #!/bin/sh
 # @configure_input@
+### BEGIN INIT INFO
+# Provides:          isdneurofile
+# Required-Start:    $remote_fs $syslog
+# Required-Stop:     $remote_fs $syslog
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+### END INIT INFO
 #
 # generated from $Id: eftd.sh.in,v 1.8 2002/01/31 18:54:38 paul Exp $
 #
