--- isdnutils-3.9.20060704+dfsg.1.orig/act2000/Makefile.in
+++ isdnutils-3.9.20060704+dfsg.1/act2000/Makefile.in
@@ -49,9 +49,9 @@
 	mkdir -p $(DESTDIR)$(MAN8DIR)
 	$(INSTALL_MAN) $< $(DESTDIR)$(MAN8DIR)/$(MANPAGE)
 
-install-data: firmware/$(FIRMWARE)
-	mkdir -p $(DESTDIR)$(DATADIR)
-	$(INSTALL_DATA) $< $(DESTDIR)$(DATADIR)/$(FIRMWARE)
+install-data: # firmware/$(FIRMWARE)
+	# mkdir -p $(DESTDIR)$(DATADIR)
+	# $(INSTALL_DATA) $< $(DESTDIR)$(DATADIR)/$(FIRMWARE)
 
 install: $(PROGRAM) install-man install-data
 	$(INSTALL_PROGRAM) $(PROGRAM) $(DESTDIR)$(SBINDIR)/$(PROGRAM)
--- isdnutils-3.9.20060704+dfsg.1.orig/eicon/Makefile.in
+++ isdnutils-3.9.20060704+dfsg.1/eicon/Makefile.in
@@ -91,8 +91,8 @@
 	$(CC) $(CFLAGS) $(INCLUDES) ./divautil/divalog.o ./divautil/loglib.o ./divautil/idilog.o ./divautil/xlog.o linuxcfg.o $(LDFLAGS) -o $(DIVALOGPROGRAM)
 
 install-data:
-	mkdir -p $(DESTDIR)$(DATA_DIR)
-	tar xpCfz $(DESTDIR)$(DATA_DIR) firmware/eicon_firm.tgz
+	# mkdir -p $(DESTDIR)$(DATA_DIR)
+	# tar xpCfz $(DESTDIR)$(DATA_DIR) firmware/eicon_firm.tgz
 
 install-man: $(PROGRAM).man $(DIVALOADPROGRAM).man $(DIVALOGPROGRAM).man $(DIVALOGDPROGRAM).man
 	mkdir -p $(DESTDIR)$(MAN8DIR)
--- isdnutils-3.9.20060704+dfsg.1.orig/icn/Makefile.in
+++ isdnutils-3.9.20060704+dfsg.1/icn/Makefile.in
@@ -48,10 +48,10 @@
 	$(CC) $(CFLAGS) $(INCLUDES) $(DEFS) $? $(LDFLAGS) -o $@
 
 install-data:
-	mkdir -p $(DESTDIR)$(DATADIR)
-	$(INSTALL_DATA) firmware/loadpg.bin $(DESTDIR)$(DATADIR)/loadpg.bin
-	$(INSTALL_DATA) firmware/pc_eu_ca.bin $(DESTDIR)$(DATADIR)/pc_eu_ca.bin
-	$(INSTALL_DATA) firmware/pc_1t_ca.bin $(DESTDIR)$(DATADIR)/pc_1t_ca.bin
+	# mkdir -p $(DESTDIR)$(DATADIR)
+	# $(INSTALL_DATA) firmware/loadpg.bin $(DESTDIR)$(DATADIR)/loadpg.bin
+	# $(INSTALL_DATA) firmware/pc_eu_ca.bin $(DESTDIR)$(DATADIR)/pc_eu_ca.bin
+	# $(INSTALL_DATA) firmware/pc_1t_ca.bin $(DESTDIR)$(DATADIR)/pc_1t_ca.bin
 
 install-man: $(PROGRAM).man
 	mkdir -p $(DESTDIR)$(MAN8DIR)
