--- a/xisdnload/Makefile.in.old	2005-08-15 09:48:31.831645000 +0200
+++ b/xisdnload/Makefile.in	2005-08-15 09:50:51.343749252 +0200
@@ -17,23 +17,28 @@
 all: xisdnload
 
 xisdnload:
-	CFLAGS= ; MAKEFLAGS= ; $(MAKE) -f Makefile EXTRA_DEFINES="$(EDEF)"
+	gcc -g -O2 -fno-strict-aliasing -I/usr/X11R6/include -c xisdnload.c
+	gcc -o xisdnload -L/usr/X11R6/lib xisdnload.o -lXaw -lXmu -lXt -lSM -lICE -lXpm -lXext -lX11
 
 config:
-	@cp Makefile.in GNUmakefile
-	@xmkmf
+	cp -p Makefile.in GNUmakefile
+	cp -p Makefile.in Makefile
 
 clean:
-	$(MAKE) -f Makefile clean
+	rm -f xisdnload xisdnload.o
 
 distclean: clean
 	rm -f Makefile Makefile.bak GNUmakefile
 
 install-man: xisdnload.man
-	$(MAKE) -f Makefile $(MANDIROPTS) install.man
+	install -d $(DESTDIR)/usr/share/man/man1
+	install -c -m 644 xisdnload.man $(DESTDIR)/usr/share/man/man1/xisdnload.1x
 
 install: xisdnload install-man
-	$(MAKE) -f Makefile $(BINDIROPT) install
+	install -d $(DESTDIR)/usr/bin
+	install -d $(DESTDIR)/etc/X11/app-defaults
+	install -c -m 0755 -s xisdnload $(DESTDIR)/usr/bin/
+	install -c -m 644 XISDNLoad.ad $(DESTDIR)/etc/X11/app-defaults/XISDNLoad
 
 uninstall:
 	rm -f $(DESTDIR)$(CONFIG_BINDIR)/xisdnload
--- xmonisdn/Makefile.in	2002-07-19 21:23:55.000000000 +0000
+++ xmonisdn/Makefile.in	2005-08-15 08:00:42.357309701 +0000
@@ -24,10 +24,10 @@
 ifeq (../.config,$(wildcard ../.config))
 	include ../.config
 	ifneq ($(CONFIG_XMONISDN_UPCMD),"")
-		I4LU_DEFINES += -DNETUP_COMMAND=\\\"'$(shell echo $(CONFIG_XMONISDN_UPCMD))'\\\"
+		I4LU_DEFINES += -DNETUP_COMMAND=\"'$(shell echo $(CONFIG_XMONISDN_UPCMD))'\"
 	endif
 	ifneq ($(CONFIG_XMONISDN_DOWNCMD),"")
-		I4LU_DEFINES += -DNETDOWN_COMMAND=\\\"'$(shell echo $(CONFIG_XMONISDN_DOWNCMD))'\\\"
+		I4LU_DEFINES += -DNETDOWN_COMMAND=\"'$(shell echo $(CONFIG_XMONISDN_DOWNCMD))'\"
 	endif
     MAN1DIROPT = MANDIR=$(CONFIG_MANDIR)/man1
     MAN3DIROPT = LIBMANDIR=$(CONFIG_MANDIR)/man3
@@ -48,7 +48,9 @@
 	mkdir -p X11/bitmaps
 	cp netinactive netwaiting netactive \
 		netactiveout netstart netstop X11/bitmaps
-	CFLAGS= ; MAKEFLAGS= ; $(MAKE) -f Makefile EXTRA_DEFINES="$(I4LU_DEFINES)"
+	gcc -g -O2 -fno-strict-aliasing $(I4LU_DEFINES) -I. -I/usr/X11R6/include -c Net.c
+	gcc -g -O2 -fno-strict-aliasing $(I4LU_DEFINES) -I. -I/usr/X11R6/include -c xmonisdn.c
+	gcc -o xmonisdn -L/usr/X11R6/lib Net.o xmonisdn.o -lXaw -lXmu -lXt -lSM -lICE -lXpm -lXext -lX11 -lm
 	rm -rf X11
 
 xmonisdn.man: xmonisdn.man.in Makefile
@@ -63,20 +65,24 @@
 	   < $< > $@
 
 config:
-	@cp Makefile.in GNUmakefile
-	@xmkmf
+	cp -p Makefile.in GNUmakefile
+	cp -p Makefile.in Makefile
 
 clean:
-	$(MAKE) -f Makefile clean
+	rm -f *.o xmonisdn
 
 distclean: clean
 	rm -f Makefile GNUmakefile Makefile.bak xmonisdn.man
 
 install-man: xmonisdn.man
-	$(MAKE) -f Makefile $(MANDIROPTS) install.man
+	install -d $(DESTDIR)/usr/X11R6/man/man1
+	install -c -m 644 xmonisdn.man $(DESTDIR)/usr/X11R6/man/man1/xmonisdn.1x
 
 install: xmonisdn install-man
-	$(MAKE) -f Makefile $(BINDIROPT) install.all
+	install -d $(DESTDIR)/usr/bin
+	install -d $(DESTDIR)/usr/X11R6/include/X11/bitmaps
+	install -c -m 0755 -s xmonisdn $(DESTDIR)/usr/bin/
+	install -c -m 644 netinactive netwaiting netactive netactiveout netstart netstop  $(DESTDIR)/usr/X11R6/include/X11/bitmaps
 
 uninstall:
 	rm -f $(DESTDIR)$(CONFIG_BINDIR)/xmonisdn
