--- isdnutils-3.7.2005-07-09.orig/Makefile
+++ isdnutils-3.7.2005-07-09/Makefile
@@ -200,9 +200,12 @@
 subconfig: scripts/autoconf.h
 	@echo Selected subdirs: $(BUILD_ONLY) $(SUBDIRS)
 	@set -e; for i in `echo $(BUILD_ONLY) $(SUBDIRS)`; do \
-		if [ -x $$i/configure ] ; then \
+		if [ $$i = eicon ]; then \
 			/bin/echo -e "\nRunning configure in $$i ...\n"; sleep 1; \
-			(cd $$i; ./configure --sbindir=$(CONFIG_SBINDIR) --bindir=$(CONFIG_BINDIR) --mandir=$(CONFIG_MANDIR) --datadir=$(CONFIG_DATADIR) || $(MAKE) -C ../ ERRDIR=$$i cfgerror); \
+			(cd $$i; echo ./configure --with-sbin=$(CONFIG_SBINDIR) --bindir=$(CONFIG_BINDIR) --with-man=$(CONFIG_MANDIR) --with-firmware=$(CONFIG_DATADIR); CONFIG_CARD_SBINDIR=$(CONFIG_SBINDIR) ./configure --with-sbin=$(CONFIG_SBINDIR) --bindir=$(CONFIG_BINDIR) --with-man=$(CONFIG_MANDIR) --with-firmware=$(CONFIG_DATADIR) || $(MAKE) -C ../ ERRDIR=$$i cfgerror); \
+		elif [ -x $$i/configure ] ; then \
+			/bin/echo -e "\nRunning configure in $$i ...\n"; sleep 1; \
+			(cd $$i; echo ./configure --sbindir=$(CONFIG_SBINDIR) --bindir=$(CONFIG_BINDIR) --mandir=$(CONFIG_MANDIR) --datadir=$(CONFIG_DATADIR); ./configure --sbindir=$(CONFIG_SBINDIR) --bindir=$(CONFIG_BINDIR) --mandir=$(CONFIG_MANDIR) --datadir=$(CONFIG_DATADIR) || $(MAKE) -C ../ ERRDIR=$$i cfgerror); \
 		elif [ -f $$i/Makefile.in ] ; then \
 			/bin/echo -e "\nRunning make -f Makefile.in config in $$i ...\n"; sleep 1; \
 			$(MAKE) -C $$i -f Makefile.in config; \
