--- isdnutils-3.3.orig/vbox/utils/vboxplay.in
+++ isdnutils-3.3/vbox/utils/vboxplay.in
@@ -44,7 +44,9 @@
 		# if anyone has a better suggestion,
 		# send email to isdnutils@packages.debian.org!
                 # thankfully, aplay understands '-' to mean stdin
-		DOPLAY="$PLAY -q -u -"
+                # 20021115 removed -u option due to #165757,
+                # aplay doesn't understand that option (anymore?)
+		DOPLAY="$PLAY -q -"
 	fi
 fi
 if [ ! -x $PLAY ]; then	# use play (from sox) instead
