--- a/vbox/Makefile.in	2005-01-08 17:23:43.000000000 +0100
+++ b/vbox/Makefile.in~	2005-01-08 17:22:59.000000000 +0100
@@ -53,7 +53,7 @@
 #----------------------------------------------------------------------------#
 
 CC		= @CC@
-CFLAGS		= @CFLAGS@ @LINK_NCURSES_INCL@ @LINK_TCL_INCL@
+CFLAGS		= @CFLAGS@ @LINK_NCURSES_INCL@ @LINK_TCL_INCL@ -Wall -fno-strict-aliasing
 LDFLAGS		= @LDFLAGS@
 
 #----------------------------------------------------------------------------#
