Index: isdnutils-3.9.20060704+dfsg.1/vbox/src/vboxd.c
===================================================================
--- isdnutils-3.9.20060704+dfsg.1.orig/vbox/src/vboxd.c	2010-07-27 21:24:00.005877529 +0200
+++ isdnutils-3.9.20060704+dfsg.1/vbox/src/vboxd.c	2010-07-27 21:25:33.485880578 +0200
@@ -389,7 +389,7 @@
 	fd_set         rmask;
 	char          *stop;
 	int            p;
-	int            c;
+	char           c;
 	int            rc;
 
 	*line = '\0';
